### PR TITLE
replace markdown-spellchecker with yaspeller

### PR DESCRIPTION
**Pull request by shannonwells**
Problem
=======
Dependabot alerts were bugging me like crazy.

Solution
========
`markdown-spellchecker` is the package that depends on outdated & insecure versions of `marked`, and it appears to be no longer maintained.  I searched a bit for an alternative and found one that is actively maintained, [`yaspeller`](https://github.com/hcodes/yaspeller).

Some advantages of yaspeller:
You can tell it to ignore any words with numbers in them, words with all uppercase, and its custom dictionary also accepts regular expressions. This made our dictionary a lot smaller.  It also uses a config file.

Change summary:
---------------
- [x] install yaspeller and uninstall markdown-spellcheker
- [x] configure yaspeller and convert the dictionary to its format
- [x] delete the old dictionary
- [x] update the package.json script for running the spellchecker
- [x] fix any newly discovered spelling mistakes
- [x] make sure tests/CI still pass

Steps to Verify:
----------------
- CI should pass, but if you want to run it yourself, just check out the branch, run `npm install` and then `npm run spellcheck`, and there should be no errors.

Screenshots (optional):
-----------------------
```zsh
shannonwells github/spec: chore/dependabot-marked $ npm run spellcheck

> liberty-protocol-spec@0.0.1 spellcheck
> yaspeller .

Spelling check:
✓ /Users/shannonwells/github/spec/.github/ISSUE_TEMPLATE/DSNP_Issue.md 759 ms
✓ /Users/shannonwells/github/spec/CODE_OF_CONDUCT.md 783 ms
✓ /Users/shannonwells/github/spec/.github/ISSUE_TEMPLATE/DSNP_Discussion.md 806 ms
✓ /Users/shannonwells/github/spec/.github/PULL_REQUEST_TEMPLATE.md 813 ms
✓ /Users/shannonwells/github/spec/.github/ISSUE_TEMPLATE/DSNP_Improvement_Proposal.md 826 ms
✓ /Users/shannonwells/github/spec/CONTRIBUTING.md 413 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Associated/Location.md 556 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Associated/Attachments.md 596 ms
✓ /Users/shannonwells/github/spec/README.md 606 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Associated/Hash.md 588 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Associated/Tag.md 425 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Types/Profile.md 389 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Types/Note.md 398 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Announcements.md 393 ms
✓ /Users/shannonwells/github/spec/pages/ActivityContent/Overview.md 454 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/BatchPublications.md 463 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Graph.md 419 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Overview.md 378 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Identifiers.md 450 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Identity.md 442 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Serializations.md 455 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Types/Broadcast.md 428 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Signatures.md 458 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Types/Profile.md 454 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Types/GraphChange.md 467 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Types/Reaction.md 548 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Types/Reply.md 441 ms
✓ /Users/shannonwells/github/spec/pages/DSNP/Types/Tombstone.md 417 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/Identifiers.md 435 ms
✓ /Users/shannonwells/github/spec/pages/Draft/Archivists.md 494 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/Overview.md 384 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/Identity.md 465 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/IdentityFactory.md 464 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/Publishing.md 455 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/Registry.md 446 ms
✓ /Users/shannonwells/github/spec/pages/Ethereum/Validation.md 439 ms
✓ /Users/shannonwells/github/spec/pages/Implementations.md 582 ms
✓ /Users/shannonwells/github/spec/pages/Index.md 582 ms
✓ /Users/shannonwells/github/spec/pages/Reference/Glossary.md 532 ms
✓ /Users/shannonwells/github/spec/pages/Roadmap.md 492 ms
✓ /Users/shannonwells/github/spec/pages/SUMMARY.md 441 ms
No errors.
Checking finished: 4.442 sec.
```

### CODE REVIEW

File: .yaspellerrc.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
+{
```
Comment (shannonwells) [inline .yaspellerrc.json:1]: note how the config file which includes the dictionary is 44 lines shorter than the old dictionary.

File: pages/DSNP/Types/GraphChange.md [hunk 2]
```diff
@@ -70,7 +70,7 @@ For example:
 Each graph change event represents a state transition for the graph.
 The state of the graph at any time is given by taking the state of the graph at a previous time and applying all graph change events not previously applied in the order specified above.
 Once those graph change events are retrieved, they can be ordered to reflect the current graph state
-(i.e. Charlie has followed Bob then unfollowed him and then followed him again. The graph state reflects that Charlie is Following Bob).
+(i.e. Charlie has followed Bob, then he unfollowed him, and then followed him again. The graph state reflects that Charlie is Following Bob).
```
Comment (shannonwells) [inline pages/DSNP/Types/GraphChange.md:73]: for some reason yaspeller errored on "then unfollowed" so I just tweaked it.

File: pages/Ethereum/IdentityFactory.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ Official implementation contract addresses will be published once deployed.
 
 ### Clone Interface
 
-Clones follow [EIP 1167](https://eips.ethereum.org/EIPS/eip-1167) for a **non-upgradeable** identity contract.
+Clones follow [EIP 1167](https://eips.ethereum.org/EIPS/eip-1167) for a **non-upgradable** identity contract.
```
Comment (shannonwells) [inline pages/Ethereum/IdentityFactory.md:65]: 🤷‍♀️

File: pages/Ethereum/Overview.md [hunk 4]
```diff
@@ -15,10 +15,11 @@ Other versions may be available in the [official contracts code repository](http
 Official DSNP interfaces, contracts, and deployment information may be found in [GitHub](https://github.com/LibertyDSNP/contracts).
 
 ## Libraries
-
+<!-- yaspeller ignore:start -->
```
Comment (shannonwells) [inline pages/Ethereum/Overview.md:18]: and if yaspeller is being unreasonable this is how you tell it to skip something.

File: pages/Index.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ Here you can find the detailed specification documentation for DSNP, officiation
 ## Goals & Purpose
 
 The free communication of users on the internet faces a variety of problems in the modern day.
-These challenges include censorship from state and corporate actors, the amplification of misinformation through viral content, and an ever-shrinking collection of near monopolies with absolute power over social interaction in the 21st century.
+These challenges include censorship from state and corporate actors, the amplification of misinformation through viral content, and an ever-shrinking collection of near monopolies with absolute power over social interaction in the twenty-first century.
```
Comment (shannonwells) [inline pages/Index.md:9]: WHATEVER, YASPELLER

File: .yaspellerrc.json [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  "excludeFiles": [
+    ".git",
+    "yaspeller",
```
Comment (mehtaishita) [inline .yaspellerrc.json:4]: why does it need 'yaspeller' as part of its exclude?

**Review by mehtaishita — APPROVED** [review]

**Comment by mehtaishita** [comment]
I checked out the branch and ran the spell check locally and it works fine for me

File: .yaspellerrc.json (hunk 6 above)
Comment (shannonwells) [reply .yaspellerrc.json:4]: ¯\_(ツ)_/¯

**Pull request merged**